src/Core/Services/Locator.cpp

```diff
@@ -0,0 +1,23 @@
+#include "Locator.hpp"
+
+// Static variable default values
+std::unique_ptr<ArchHooks> Locator::archHooks = nullptr;
+std::unique_ptr<Core::ILogger> Locator::logger = nullptr;
+
+ArchHooks *Locator::getArchHooks() {
+    return archHooks.get();
+}
+
+Core::ILogger *Locator::getLogger() {
+    return logger.get();
+}
+
+void Locator::provide(ArchHooks *hooks) {
+    if(!Locator::archHooks)
+        Locator::archHooks = std::unique_ptr<ArchHooks>(hooks);
```

Comment:
```suggestion
        Locator::archHooks = std::make_unique<ArchHooks>(hooks);
```

Reply:
I tried this change, but it won't compile, and I don't think it's what I want to do here. I'm trying to convert a raw pointer into a unique_ptr, instead of creating a new object that is unique in the first place.